public/app/features/trails/MetricSelectScene.tsx

```diff
@@ -319,3 +320,26 @@ function getStyles(theme: GrafanaTheme2) {
     }),
   };
 }
+
+// Consider any sequence of characters not permitted for metric names as a sepratator
+const splitSeparator = /[^a-z0-9_:]+/;
+
+function createSearchRegExp(spaceSeparatedMetricNames?: string) {
+  if (!spaceSeparatedMetricNames) {
+    return null;
+  }
+  const searchParts = spaceSeparatedMetricNames
+    ?.toLowerCase()
+    .split(splitSeparator)
+    .filter((part) => part.length > 0)
+    .map((part) => `(?=(.*${part}.*))`);
+
+  if (searchParts.length === 0) {
+    return null;
+  }
+
+  const regex = searchParts.join('');
+  //  (?=(.*expr1.*))(?=().*expr2.*))...
+  // The ?=(...) lookahead allows us to match these in any order.
+  return new RegExp(regex, 'igy');
+}
```

Comment:
grafana/ui exports this util function that does fuzzy matching, maybe worth a try? https://github.com/grafana/grafana/blob/main/packages/grafana-ui/src/utils/fuzzy.ts#L27

Reply:
iirc that one doesnt work out of order, and is rather slow (though i dont think perf matters here?)

i was planning on eventually replacing its innards with a call into uFuzzy :)

Reply:
I can see this evolving to use uFuzzy instead because it seems to be a closer match to what we want based on what @leeoniya has indicated.
But for now, being fuzzy isn't truly the goal, so I have changed the PR name accordingly.
We currently want something comparable to the prometheus metric selector.
Let's plan to revisit this next Q.